test/tools/yulopti.cpp

```diff
@@ -291,8 +317,33 @@ Allowed options)",
 		return 1;
 	}
 
-	if (arguments.count("input-file"))
-		YulOpti{}.runInteractive(input);
+	if (nonInteractive || arguments.count("input-file"))
+	{
+		YulOpti yulOpti;
+		bool disambiguated = false;
+		if (!nonInteractive)
+			cout << input << endl;
+		if (arguments.count("steps"))
+		{
+			string sequence = arguments["steps"].as<string>();
+			try
+			{
+				OptimiserSuite::validateSequence(sequence);
+			}
+			catch(std::exception const& _exception)
```

Comment:
`OptimizerException` is the only exception that can really be considered a validation error. Anything else is just a random failure. Especially, right now it silences our asserts.

```suggestion
			catch (OptimizerException const& _exception)
```

Reply:
I'm not sure - there's also ``Unhandled syntactic element in the abbreviation sequence``, which throws a ``YulAssertion`` and to me it seems preferable to just catch and print it here than to crash...

Reply:
This one is an assert on purpose. It's no a runtime error. It can only happen if you add a new special char to [`NonStepAbbreviations`](https://github.com/ethereum/solidity/blob/a06cb0929ff264af483d914edaff2b346e165ee8/libyul/optimiser/Suite.h#L54) and then forget to handle it here.